rust/db_handling/src/identities.rs

```diff
@@ -280,13 +280,39 @@ pub fn create_increment_set(increment: u32, multisigner: &MultiSigner, network_s
 }
 
 /// Check derivation format and determine whether there is a password
-pub fn check_derivation_format(path: &str) -> Result<bool, ErrorSigner> {
+pub fn is_passworded(path: &str) -> Result<bool, ErrorSigner> {
     match REG_PATH.captures(path) {
         Some(caps) => Ok(caps.name("password").is_some()),
         None => return Err(ErrorSigner::AddressGeneration(AddressGeneration::Extra(ExtraAddressGenerationSigner::InvalidDerivation))),
     }
 }
 
+/// Enum to describe derivation proposed
+pub enum DerivationCheck {
+    BadFormat, // bad format, still gets json back into UI
+    Password, // passworded, no dynamic checking for collisions
+    NoPassword(Option<(MultiSigner, AddressDetails)>), // no password and optional collision
+}
```

Comment:
Comments for enum variants should be actually doc comments that will go into the documentation, see for instance [Result](https://doc.rust-lang.org/std/result/enum.Result.html) docs and comments in the source code.

Reply:
Yes, these are really quick stubs that should be replaced with proper docs on documentation pass. I'll note these specific things in issues to keep track of them.